src/instruments/src/Common/gauges.tsx

```diff
@@ -171,3 +174,119 @@ export const VerticalSegment: FC<VerticalSegmentProps> = ({ x, y, height, rangeS
+    return (decimalSplit);
+}
+
+export function valueRadianAngleConverter(value, min, max, endAngle, startAngle) {
+    const valuePercentage = (value - min) / (max - min);
+    let angleInRadians = startAngle > endAngle
+        ? startAngle + (valuePercentage * (360 - startAngle + endAngle)) - 90
+        : startAngle + (valuePercentage * (endAngle - startAngle)) - 90;
+    angleInRadians *= (Math.PI / 180.0);
+    return ({
+        x: Math.cos(angleInRadians),
+        y: Math.sin(angleInRadians),
+    });
+}
+
+type GaugeMarkerComponentType = {
+    value: number,
+    x: number,
+    y: number,
+    min: number,
+    max: number,
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    className: string,
+    showValue?: boolean,
+    indicator?: boolean,
+    outer?: boolean,
+    multiplier?: number,
+};
+
+const GaugeMarkerComponentNoMemo = ({ value, x, y, min, max, radius, startAngle, endAngle, className, showValue, indicator, outer, multiplier } : GaugeMarkerComponentType) => {
+    const dir = valueRadianAngleConverter(value, min, max, endAngle, startAngle);
+
+    if (typeof multiplier === 'undefined') multiplier = 1.15;
+
+    let start = {
+        x: x + (dir.x * radius * 0.9),
+        y: y + (dir.y * radius * 0.9),
+    };
+    let end = {
+        x: x + (dir.x * radius),
+        y: y + (dir.y * radius),
+    };
+
+    if (outer) {
+        start = {
+            x: x + (dir.x * radius),
+            y: y + (dir.y * radius),
+        };
+        end = {
+            x: x + (dir.x * radius * multiplier),
+            y: y + (dir.y * radius * multiplier),
+        };
+    }
+
+    if (indicator) {
+        start = { x, y };
+
+        end = {
+            x: x + (dir.x * radius * multiplier),
+            y: y + (dir.y * radius * multiplier),
+        };
+    }
+
+    // Text
+    const multiplierX = value >= 8 ? 0.8 : 0.7;
+    const multiplierY = value >= 8 ? 0.52 : 0.7;
+    const pos = {
+        x: x + (dir.x * (radius * multiplierX)),
+        y: y + (dir.y * (radius * multiplierY)),
+    };
+
+    const textValue = !showValue ? '' : Math.abs(value).toString();
+
+    return (
+        <>
+            <line x1={start.x} y1={start.y} x2={end.x} y2={end.y} className={className} />
+            <text x={pos.x} y={pos.y} className={className} alignmentBaseline="central" textAnchor="middle">{textValue}</text>
+        </>
+    );
+};
+export const GaugeMarkerComponent = memo(GaugeMarkerComponentNoMemo);
+
+type GaugeComponentProps = {
+    x: number,
+    y: number,
+    radius: number,
+    startAngle: number,
+    endAngle: number,
+    className: string,
+    manMode: boolean,
+}
+
+export const GaugeComponentNoMemo: FC<GaugeComponentProps> = ({ x, y, radius, startAngle, endAngle, className, children, manMode }) => {
+    const d = describeArc(x, y, radius, startAngle, endAngle);
+
+    return (
+        <>
+            <g id="HideOrShowGauge" className={manMode ? 'Show' : 'Hide'}>
```

Comment:
**issue:** this seems to be presumably used in multiple places, therefore a `class` should be used and not an `id`.
**suggestion:** `.HideOrShowGauge` seems to handle the `GaugeComponent` container as a whole, so it should be named appropriately, like `.GaugeComponent`